test/check-application

```diff
@@ -3005,6 +3026,65 @@ class TestApplication(testlib.MachineCase):
         else:
             self.assertNotIn("Podman", b.text("#host-apps"))
 
+    @testlib.skipImage("nginx not installed", "centos-*", "rhel-*", "debian-*", "ubuntu-*", "arch")
+    @testlib.skipOstree("nginx not installed")
+    def testNonSearchableRegistry(self):
+        b = self.browser
+        m = self.machine
+
+        self.setupRegistry()
+
+        # Nginx config simulates behavior of ghcr.io in terms of `podman search` and `podman manifest inspect`
+        self.execute(True, f"""
+            echo '{NGINX_DEFAULT_CONF}' > /etc/nginx/conf.d/default.conf
+            setsebool -P httpd_can_network_connect 1
+            systemctl start nginx.service
+        """)
+        self.addCleanup(m.execute, """
+            systemctl stop nginx.service
+            setsebool -P httpd_can_network_connect 0
+            rm -f /etc/nginx/default.conf
+        """)
+
+        container_name = "localhost:80/my-busybox"
+        m.execute(f"podman tag {IMG_BUSYBOX} {container_name}; podman push {container_name}")
+        # Untag busybox image which duplicates the image we are about to download
+        m.execute(f"podman rmi -f {IMG_BUSYBOX} localhost:80/my-busybox")
+
+        self.login()
+
+        b.click("#containers-containers button.pf-v5-c-button.pf-m-primary")
+        b.set_input_text("#run-image-dialog-name", "new-busybox")
+        # Searching for container by prefix fails
+        b.set_input_text("#create-image-image-select-typeahead", "localhost:80/my-busy")
+        b.wait_text("button.pf-v5-c-select__menu-item.pf-m-disabled", "No images found")
+        b.wait_in_text(".pf-v5-c-alert.pf-m-danger", "couldn't search registry")
+
+        # Giving full image name finds valid manifest therefore image is pullable
+        b.set_input_text("#create-image-image-select-typeahead", container_name)
+        b.wait_text("button.pf-v5-c-select__menu-item:not(.pf-m-disabled)", container_name)
+        # Select image and create a container
+        b.click(f"button.pf-v5-c-select__menu-item:contains({container_name})")
+        b.click("#create-image-create-btn")
+
+        # Wait for download to finish
+        sel = "span:not(.downloading)"
+        b.wait(lambda: self.getContainerAttr(container_name, "State", sel) in "Created")
+
+        b.click("#containers-containers button.pf-v5-c-button.pf-m-primary")
+        # "couldn't search registry" error is hidden when some local image is found
+        b.set_input_text("#create-image-image-select-typeahead", "localhost:80/my-busy")
+        b.wait_not_present(".pf-v5-c-alert.pf-m-danger")
```

Comment:
You need a positive check before a `wait_not_present()` call, as this is always immediately true. I.e. wait for the found local container to be displayed first.